include/Row.hpp

```diff
@@ -228,24 +230,33 @@ class Row {
 
     /*
      * Retrieve the value at the specified column index as a geographical point.
-     * This data type is not currently supported by the C++ client so this always
-     * throws an exception.
-     * @throws UnsupportedTypeException always
-     * @return never returns
+     *
+     * @throws InvalidColumnException The index of the column was invalid or the type of the column does
+     * not match the type of the get method.
+     * @return the deserialized GeographyPoint value.
      */
-    void getGeographyPoint(int32_t column) {
-        throw UnsupportedTypeException(wireTypeToString(WIRE_TYPE_GEOGRAPHY_POINT));
+    GeographyPoint getGeographyPoint(int32_t column) throw(voltdb::InvalidColumnException) {
+        validateType(WIRE_TYPE_GEOGRAPHY_POINT, column);
+        GeographyPoint gpoint;
+        m_wasNull = false;
+        gpoint.deserializeFrom(m_data, getOffset(column), m_wasNull);
+        return gpoint;
```

Comment:
I don't understand what you mean here.

1. It's two doubles.  It's hardly more expensive than returning a single double.  They will probably be returned in registers, or else they will be returned by reference.  Structures which have only double precision members are often returned specially by returning values in registers.  It's something having to do with optimizing complex, quaternion and octernion results.

I have changed this so that the constructor is a call to deserializeFrom.  I'm not sure this is a good idea, though, since it seems like a premature optimization.